### PR TITLE
✨Introduce tree shaking

### LINKED ISSUE #1043: Support for tree-shaking in eds-core-react package
Commonjs does not support tree-shaking when deconstructing values from a provided object.

This means that when, for example, `import { Button } from '@equinor/eds-core-react'` in a commojs project, it will import all the other components in the library regardless. 

Since Typescript still uses `commonjs` for modules, we should look into providing named paths for components.

Example of relative path: `import Button from '@equinor/eds-core-react/Button'`.

- [ ] Research/Analyse on how to solve this
- [ ] Check wether this can configured in rollup or not.

Notes: 
- https://material-ui.com/guides/minimizing-bundle-size/#when-and-how-to-use-tree-shaking
- https://rollupjs.org/guide/en/#outputpreservemodules

### PR DESCRIPTION
**Pull request by oddvernes**
Resolves #1043

### CODE REVIEW

**Comment by oddvernes** [comment]
There is an issue with the focus-visible polyfill, which is imported in the index.ts file. It breaks the focus outline wherever this is used, such as radio/checkbox/tab

**Comment by mimarz** [comment]
> There is an issue with the focus-visible polyfill, which is imported in the index.ts file. It breaks the focus outline wherever this is used, such as radio/checkbox/tab

Guess it makes sense with the support for tree-shaking now, it gets thrown away as there are no clear indications for the module loarder that the polyfill is being used anywhere 🤔

**Comment by oddvernes** [comment]
More importantly the current build is crashing with "React is not defined" if you try to run it instead of just building it in an external project like create-react-app or codesandbox

**Comment by oddvernes** [comment]
After som testing, `import 'focus-visible'` must be done in a file (any file) that is used in the project, and then it will be included as part of the bundle downstream like so: 
bilde
It can not be put in an index file, so there isn't really any super logical place to do it, but it can be done anywhere as long as it is in a component or file that is imported and used by any component.

I think we should ditch this polyfill the moment it is introduced in safari (currently can be activated as an 'experimental feature'), as it kind of interferes with any focusable element in the dom, not just eds components, by toggling 'focus-visible' class and also a data-attribute

**Comment by mimarz** [comment]
> I think we should ditch this polyfill the moment it is introduced in safari (currently can be activated as an 'experimental feature'), as it kind of interferes with any focusable element in the dom, not just eds components, by toggling 'focus-visible' class and also a data-attribute

Agreed, I think this has been the plan all along, we're just waiting for that someone special to accept reality 😅

**Comment by oddvernes** [comment]
> 
> Agreed, I think this has been the plan all along, we're just waiting for that someone special to accept reality 😅

The [status as of may 2021](https://blogs.igalia.com/mrego/2021/06/07/focus-visible-in-webkit-may-2021/)

**Review by mimarz — APPROVED** [review]
Just a small clarification in README, else it looks nice! 👍

File: libraries/core-react/package.json [hunk 1]
```diff
@@ -2,11 +2,11 @@
   "name": "@equinor/eds-core-react",
   "version": "0.15.0",
   "description": "The React implementation of the Equinor Design System",
-  "main": "src/index.ts",
+  "sideEffects": false,
+  "main": "dist/core-react.cjs.js",
   "publishConfig": {
```
[the remark this answers is not in the document]
Comment (oddvernes) [reply libraries/core-react/package.json:7]: Is that not still used? It is the only place module is defined

[the remark this answers is not in the document]
Comment (mimarz) [reply libraries/core-react/package.json:7]: I think it was to support when storybook was outside the core-react library folder so there was a seperate bundle with sourcemaps and some more stuff just for storybook. The paths are now the same and we don't have "special" bundles anymore. Do you remember @vnys ?

File: README.md [hunk 2]
```diff
@@ -114,6 +111,15 @@ If you use Typescript, make sure you have typescript >= 3.8 as a devDependency:
 ```sh
 npm install typescript --save-dev
 ```
+* _Note_ from version `0.16.0` we added native support for focus-visible (enable focus-ring on elements when using <kbd>TAB</kbd> key) and removed the [polyfill](https://github.com/WICG/focus-visible). Optionally, if your project requires support for keyboard navigation with Safari, you now need to include the polyfill yourself:
```
[the remark this answers is not in the document]
Comment (oddvernes) [reply README.md]: I added sub-heading and rewrote it a little bit

**Pull request merged**